lib/haml_lint/tree/tag_node.rb

```diff
@@ -20,6 +20,15 @@ def contains_script?
       @value[:parse] && !@value[:value].strip.empty?
     end
 
+    # Returns whether this tag has a specified attribute
+    # `=`).
+    #
+    # @return [true,false]
+    def has_hash_attribute?(attribute)
+      hash_attributes? &&
+        hash_attributes_source.scan(/(#{attribute})\W/).any?
```

Comment:
I don't think I understand what you mean by the `parser` gem not giving access to hash keys as nodes. If you look at the implementation at https://github.com/causes/haml-lint/blob/master/lib/haml_lint/linter/class_attribute_with_static_value.rb#L43 you can see we're checking to see if a key is a string or symbol.